shell/server/identity.js

```diff
@@ -57,9 +58,72 @@ makeIdentity = (identityId, requriements) => {
 
 globalFrontendRefRegistry.register({
   frontendRefField: "identity",
+  typeId: Identity.typeId,
 
   restore(db, saveTemplate, identityId) {
     return new Capnp.Capability(new IdentityImpl(db, saveTemplate, identityId),
                                 IdentityRpc.PersistentIdentity);
   },
+
+  validate(db, session, identityId, options) {
```

Comment:
I'm concerned that this `options` isn't being type-checked.

More generally, possibly validate() has the wrong signature. It should input a "frontendRef request", which may or may not look the same as a final frontendRef. It should then return -- in addition to `descriptor` and `requirements` -- the final ref.

This relates to why validate() is documented as being allowed to modify the frontendRef passed in -- some other ref types really send up more of a "request" which validate() munges into the final form. But you can't quite do that here since an identity ref is just a string, not an object, so it's not something you can modify in-place.